### ISSUE TITLE
build failures with `ghc-7.0.1`

### ISSUE THREAD

**Issue by soenkehahn** [lpsmith/bytestring-builder#9]
Current master fails to compile with `ghc-7.0.1` with the error included at the bottom.

Looks like it's a bit nasty to fix, but it'd be nice if `bytestring-builder` would compile with that version of `ghc`.

```
Building bytestring-builder-0.10.8.1.0...
Preprocessing library bytestring-builder-0.10.8.1.0...
[ 9 of 13] Compiling Data.ByteString.Builder.Internal ( src/Data/ByteString/Builder/Internal.hs, dist/build/Data/ByteString/Builder/Internal.o )
ghc: panic! (the 'impossible' happened)
  (GHC version 7.0.1 for x86_64-unknown-linux):
        dsLet: unlifted
    AbsBinds
    [b{tv a2jk} [sk]]
    []
    [ip'{v a22y} [lid] <= [b{tv a2jk} [sk]] ip'{v a2jj} [lid]]
      ip'{v a22y} [lid]
        :: forall b{tv a2jk} [sk]. base:GHC.Ptr.Ptr{tc 33A} b{tv a2jk} [sk]
      [LclId]
      { {831:15-46}
        {831:15-46}
        !((ip'{v a2jj} [lid] :: base:GHC.Ptr.Ptr{tc 33A} b{tv a2jk} [sk]))
          = {831:22-46}
            {831:25-33}
            (base:GHC.Ptr.plusPtr{v rdH} [gid]) @ base:GHC.Word.Word8{tc 32U}
            @ b{tv a2jk} [sk]
              ip{v a22s} [lid] outRemaining{v a22v} [lid] }
    EvBinds{{}}
    base:GHC.Base.${v 019} [gid]
      @ (bytestring-builder-0.10.8.1.0:Data.ByteString.Builder.Internal.BuildSignal{tc r1Yb}
           a{tv a2iU} [sk])
      @ (ghc-prim:GHC.Types.IO{tc 32I}
           (bytestring-builder-0.10.8.1.0:Data.ByteString.Builder.Internal.BuildSignal{tc r1Yb}
              a{tv a2iU} [sk]))
      (base:GHC.Base.return{v 01T} [gid[ClassOp]]
         @ <nt>ghc-prim:GHC.Types.IO{tc 32I}
         $dMonad{v a2jl} [lid]
         @ (bytestring-builder-0.10.8.1.0:Data.ByteString.Builder.Internal.BuildSignal{tc r1Yb}
              a{tv a2iU} [sk]))
      (bytestring-builder-0.10.8.1.0:Data.ByteString.Builder.Internal.bufferFull{v r1YP} [lid]
         @ a{tv a2iU} [sk]
         (ghc-prim:GHC.Types.I#{(w) v 6d} [gid[DataCon]] 1)
         ope{v a22u} [lid]
         (go{v a2iW} [lid]
            (ip'{v a22y} [lid] @ base:GHC.Word.Word8{tc 32U})))

Please report this as a GHC bug:  http://www.haskell.org/ghc/reportabug
```

**Comment by lpsmith** [comment]
Odd.  Does the corresponding version of bytestring also cause GHC 7.0.1 to crash?

**Comment by soenkehahn** [comment]
Yes, `bytestring-0.10.8.1` also fails with a very similar message (I think differing only in generated names) in `Data.ByteString.Builder.Internal`.

**Comment by soenkehahn** [comment]
Although here we have a successful build: http://matrix.hackage.haskell.org/package/bytestring
I don't know why.

**Edit**: Probably because the build is using `7.0.2` or newer where this bug seems to be fixed. So I guess the question is whether `bytestring` and `bytestring-builder` are aiming to support `ghc-7.0.1`, even though that would mean working around a bug that was fixed in `7.0.2`. I'm also not sure what it'd take to work around that bug.

**Comment by lpsmith** [comment]
To be honest, this package is mostly just a .cabal file and a shell script.   So the proper thing to do would be to fix it in bytestring,  then ping me to regenerate this package.

Also, how much work is this really worth?  GHC 7.0.1 might be interesting in a study of say,  GHC performance over time and whatnot,  but I doubt it is of any interest to the vast majority of people anymore.   I myself am still using GHC 7.6 for a few things,  but am quickly losing interest in anything less than 7.8

**Comment by soenkehahn** [comment]
I see. Closing then.

**Comment by RyanGlScott** [comment]
The fact that `bytestring-builder` doesn't build on GHC 7.0.1 is kind of annoying for me, since it causes the Travis builds for `base-compat` and `base-orphans` to fail. Luckily, the fix turns out to be pretty simple (see #10).
